src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/model/CaseRecordFactory.java

```diff
@@ -0,0 +1,44 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.model;
+
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.model.caserecord.ExceptionRecord;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.model.caserecord.SupplementaryEvidenceRecord;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.model.Envelope;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.model.CaseRecord.Construct;
+import static uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.model.CaseRecord.Record;
+
+public final class CaseRecordFactory {
+
+    private static final Map<Record, Construct> AVAILABLE_MODELS = new HashMap<>();
+
+    static {
+        AVAILABLE_MODELS.put(Record.EXCEPTION_RECORD, ExceptionRecord::new);
+        AVAILABLE_MODELS.put(Record.SUPPLEMENTARY_EVIDENCE, SupplementaryEvidenceRecord::new);
+    }
+
+    public static CaseDataCreator getCaseDataCreator(Envelope envelope) {
+        Construct caseRecordConstruct;
+
+        switch (envelope.classification) {
```

Comment:
why not just get it from the map directly ?
AVAILABLE_MODELS.get(envelope.classification).appy(evelope) ?

Reply:
Classification differ. And anyway, this is entry point to decide which record to create. not only envelope classification decides that